exporter/pbm_collector.go

```diff
@@ -0,0 +1,191 @@
+// mongodb_exporter
+// Copyright (C) 2024 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporter
+
+import (
+	"context"
+	"time"
+
+	"github.com/percona/percona-backup-mongodb/sdk"
+	"github.com/percona/percona-backup-mongodb/sdk/cli"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// pbm collector collects metrics from PBM (Percona Backup for MongoDb).
+type pbmCollector struct {
+	ctx      context.Context
+	mongoURI string
+	base     *baseCollector
+}
+
+const (
+	pbmAgentStatusOK    = 0
```

Comment:
Should these be based on iota? 
For reference: https://github.com/percona/pmm/blob/main/docs/process/best_practices.md?plain=1#L65